### ISSUE TITLE
cannot report intermediate result for cifar10_pytorch test on win10

### ISSUE THREAD

**Issue by 1999kevin** [microsoft/nni#3319]
**Environment**:
- NNI version: 2.0
- NNI mode (local|remote|pai): local
- Client OS: win10
- Server OS (for remote mode only):
- Python version: 3.8.3
- PyTorch/TensorFlow version: pytorch 1.7.0
- Is conda/virtualenv/venv used?: no
- Is running in Docker?: no

**What issue meet, what's expected?**:
When I try to run /trials/cifar10_pytroch on win10, I find such view on Web UI:
![IMG_0809](https://user-images.githubusercontent.com/37931674/104999928-a10cc780-5a68-11eb-9f99-a042ef16a6cd.PNG)

 As you can see, there is nothing shown in colomn metric. In my opinion, I should get intermediate result in this colomn, right? 
![IMG_0810](https://user-images.githubusercontent.com/37931674/104999965-b1bd3d80-5a68-11eb-878b-e95d81e9fcd6.JPG)

**Comment by J-shang** [comment]
Hi, could you show your experiment config, trial log and nnimanager log? You can find the logs in `~/nni-experiments/<experiment-id>`. Seems rest server doesn't receive the final result either?

**Comment by 1999kevin** [comment]
> Hi, could you show your experiment config, trial log and nnimanager log? You can find the logs in `~/nni-experiments/<experiment-id>`. Seems rest server doesn't receive the final result either?

Thank you for your reply. Here are my logs:

Config: 
{
  "revision": 440,
  "execDuration": "1h 11m 36s",
  "nextSequenceId": 9,
  "params": {
    "authorName": "default",
    "trialConcurrency": 4,
    "maxExecDuration": "6000m",
    "maxTrialNum": 10,
    "tuner": {
      "builtinTunerName": "TPE",
      "classArgs": {
        "optimize_mode": "maximize"
      },
      "checkpointDir": "C:\\Users\\a\\nni-experiments\\ePhf0tJr\\checkpoint"
    },
    "versionCheck": true,
    "clusterMetaData": [
      {
        "key": "codeDir",
        "value": "C:\\Users\\a\\Desktop\\courses\\nni\\examples\\trials\\cifar10_pytorch\\."
      },
      {
        "key": "command",
        "value": "python -u main.py"
      }
    ]
  }
}

nothing in trial log.

nnimanager log:
[2021-01-21 18:51:44] INFO [ 'Datastore initialization done' ]
[2021-01-21 18:51:44] INFO [ 'RestServer start' ]
[2021-01-21 18:51:44] INFO [ 'Construct local machine training service.' ]
[2021-01-21 18:51:44] INFO [ 'RestServer base port is 8080' ]
[2021-01-21 18:51:44] INFO [ 'Rest server listening on: http://0.0.0.0:8080' ]
[2021-01-21 18:51:47] INFO [ 'NNIManager setClusterMetadata, key: trial_config, value: {"command":"python -u main.py","codeDir":"C:\\\\Users\\\\a\\\\Desktop\\\\courses\\\\nni\\\\examples\\\\trials\\\\cifar10_pytorch\\\\.","gpuNum":1}' ]
[2021-01-21 18:51:47] INFO [ 'required GPU number is 1' ]
[2021-01-21 18:51:49] INFO [ 'Starting experiment: ePhf0tJr' ]
[2021-01-21 18:51:49] INFO [ 'Change NNIManager status from: INITIALIZED to: RUNNING' ]
[2021-01-21 18:51:49] INFO [ 'Add event listeners' ]
[2021-01-21 18:51:49] INFO [ 'Run local machine training service.' ]
[2021-01-21 18:51:50] INFO [ 'NNIManager received command from dispatcher: ID, ' ]
[2021-01-21 18:51:50] INFO [ 'NNIManager received command from dispatcher: TR, {"parameter_id": 0, "parameter_source": "algorithm", "parameters": {"lr": 0.1, "optimizer": "SGD", "model": "resnet18"}, "parameter_index": 0}' ]
[2021-01-21 18:51:50] INFO [ 'NNIManager received command from dispatcher: TR, {"parameter_id": 1, "parameter_source": "algorithm", "parameters": {"lr": 0.01, "optimizer": "Adam", "model": "resnet18"}, "parameter_index": 0}' ]
[2021-01-21 18:51:50] INFO [ 'NNIManager received command from dispatcher: TR, {"parameter_id": 2, "parameter_source": "algorithm", "parameters": {"lr": 0.1, "optimizer": "Adam", "model": "resnet18"}, "parameter_index": 0}' ]
[2021-01-21 18:51:50] INFO [ 'NNIManager received command from dispatcher: TR, {"parameter_id": 3, "parameter_source": "algorithm", "parameters": {"lr": 0.0001, "optimizer": "Adam", "model": "resnet18"}, "parameter_index": 0}' ]
[2021-01-21 18:51:54] INFO [ 'submitTrialJob: form: {"sequenceId":0,"hyperParameters":{"value":"{\\"parameter_id\\": 0, \\"parameter_source\\": \\"algorithm\\", \\"parameters\\": {\\"lr\\": 0.1, \\"optimizer\\": \\"SGD\\", \\"model\\": \\"resnet18\\"}, \\"parameter_index\\": 0}","index":0}}' ]
[2021-01-21 18:51:54] INFO [ 'submitTrialJob: form: {"sequenceId":1,"hyperParameters":{"value":"{\\"parameter_id\\": 1, \\"parameter_source\\": \\"algorithm\\", \\"parameters\\": {\\"lr\\": 0.01, \\"optimizer\\": \\"Adam\\", \\"model\\": \\"resnet18\\"}, \\"parameter_index\\": 0}","index":0}}' ]
[2021-01-21 18:51:54] INFO [ 'submitTrialJob: form: {"sequenceId":2,"hyperParameters":{"value":"{\\"parameter_id\\": 2, \\"parameter_source\\": \\"algorithm\\", \\"parameters\\": {\\"lr\\": 0.1, \\"optimizer\\": \\"Adam\\", \\"model\\": \\"resnet18\\"}, \\"parameter_index\\": 0}","index":0}}' ]
[2021-01-21 18:51:54] INFO [ 'submitTrialJob: form: {"sequenceId":3,"hyperParameters":{"value":"{\\"parameter_id\\": 3, \\"parameter_source\\": \\"algorithm\\", \\"parameters\\": {\\"lr\\": 0.0001, \\"optimizer\\": \\"Adam\\", \\"model\\": \\"resnet18\\"}, \\"parameter_index\\": 0}","index":0}}' ]
[2021-01-21 18:52:04] INFO [ 'Trial job mlpjA status changed from WAITING to RUNNING' ]
[2021-01-21 19:06:02] INFO [ 'Trial job mlpjA status changed from RUNNING to SUCCEEDED' ]
[2021-01-21 19:06:02] INFO [ 'NNIManager received command from dispatcher: TR, {"parameter_id": 4, "parameter_source": "algorithm", "parameters": {"lr": 0.01, "optimizer": "SGD", "model": "mobilenet"}, "parameter_index": 0}' ]
[2021-01-21 19:06:07] INFO [ 'submitTrialJob: form: {"sequenceId":4,"hyperParameters":{"value":"{\\"parameter_id\\": 4, \\"parameter_source\\": \\"algorithm\\", \\"parameters\\": {\\"lr\\": 0.01, \\"optimizer\\": \\"SGD\\", \\"model\\": \\"mobilenet\\"}, \\"parameter_index\\": 0}","index":0}}' ]
[2021-01-21 19:06:12] INFO [ 'Trial job WDmy0 status changed from WAITING to RUNNING' ]
[2021-01-21 19:20:15] INFO [ 'Trial job WDmy0 status changed from RUNNING to SUCCEEDED' ]
[2021-01-21 19:20:15] INFO [ 'NNIManager received command from dispatcher: TR, {"parameter_id": 5, "parameter_source": "algorithm", "parameters": {"lr": 0.001, "optimizer": "Adam", "model": "resnet18"}, "parameter_index": 0}' ]
[2021-01-21 19:20:20] INFO [ 'submitTrialJob: form: {"sequenceId":5,"hyperParameters":{"value":"{\\"parameter_id\\": 5, \\"parameter_source\\": \\"algorithm\\", \\"parameters\\": {\\"lr\\": 0.001, \\"optimizer\\": \\"Adam\\", \\"model\\": \\"resnet18\\"}, \\"parameter_index\\": 0}","index":0}}' ]
[2021-01-21 19:20:26] INFO [ 'Trial job wNJ9i status changed from WAITING to RUNNING' ]
[2021-01-21 19:34:34] INFO [ 'Trial job wNJ9i status changed from RUNNING to SUCCEEDED' ]
[2021-01-21 19:34:34] INFO [ 'NNIManager received command from dispatcher: TR, {"parameter_id": 6, "parameter_source": "algorithm", "parameters": {"lr": 0.0001, "optimizer": "Adam", "model": "resnet18"}, "parameter_index": 0}' ]
[2021-01-21 19:34:39] INFO [ 'Trial job zKpyp status changed from WAITING to RUNNING' ]
[2021-01-21 19:34:39] INFO [ 'submitTrialJob: form: {"sequenceId":6,"hyperParameters":{"value":"{\\"parameter_id\\": 6, \\"parameter_source\\": \\"algorithm\\", \\"parameters\\": {\\"lr\\": 0.0001, \\"optimizer\\": \\"Adam\\", \\"model\\": \\"resnet18\\"}, \\"parameter_index\\": 0}","index":0}}' ]
[2021-01-21 19:48:48] INFO [ 'Trial job zKpyp status changed from RUNNING to SUCCEEDED' ]
[2021-01-21 19:48:48] INFO [ 'NNIManager received command from dispatcher: TR, {"parameter_id": 7, "parameter_source": "algorithm", "parameters": {"lr": 0.001, "optimizer": "SGD", "model": "mobilenet"}, "parameter_index": 0}' ]
[2021-01-21 19:48:53] INFO [ 'Trial job Y9DPd status changed from WAITING to RUNNING' ]
[2021-01-21 19:48:53] INFO [ 'submitTrialJob: form: {"sequenceId":7,"hyperParameters":{"value":"{\\"parameter_id\\": 7, \\"parameter_source\\": \\"algorithm\\", \\"parameters\\": {\\"lr\\": 0.001, \\"optimizer\\": \\"SGD\\", \\"model\\": \\"mobilenet\\"}, \\"parameter_index\\": 0}","index":0}}' ]
[2021-01-21 19:54:17] INFO [ 'Trial job Y9DPd status changed from RUNNING to SUCCEEDED' ]
[2021-01-21 19:54:17] INFO [ 'NNIManager received command from dispatcher: TR, {"parameter_id": 8, "parameter_source": "algorithm", "parameters": {"lr": 0.01, "optimizer": "SGD", "model": "resnet18"}, "parameter_index": 0}' ]
[2021-01-21 19:54:22] INFO [ 'Trial job iufzc status changed from WAITING to RUNNING' ]
[2021-01-21 19:54:22] INFO [ 'submitTrialJob: form: {"sequenceId":8,"hyperParameters":{"value":"{\\"parameter_id\\": 8, \\"parameter_source\\": \\"algorithm\\", \\"parameters\\": {\\"lr\\": 0.01, \\"optimizer\\": \\"SGD\\", \\"model\\": \\"resnet18\\"}, \\"parameter_index\\": 0}","index":0}}' ]

**Comment by J-shang** [comment]
Hi, sorry for the late reply, we found that if we use `torch.utils.data.DataLoader` and set `num_workers>0` under windows,  `nni.report_intermediate_result()` and `nni.report_final_result()` will meet error. For a workaround, you can modify the code in `examples/trials/cifar10_pytorch/main.py`:
`trainloader = torch.utils.data.DataLoader(trainset, batch_size=128, shuffle=True, num_workers=2)` and
`testloader = torch.utils.data.DataLoader(testset, batch_size=100, shuffle=False, num_workers=2)` to
`trainloader = torch.utils.data.DataLoader(trainset, batch_size=128, shuffle=True)` and
`testloader = torch.utils.data.DataLoader(testset, batch_size=100, shuffle=False)`

We will continue to track the root cause of this problem, thank you for your issue : )

**Comment by 1999kevin** [comment]
Thanks a lot for your reply. This change works for me!
